plugins/modules/mso_schema_template_service_graph.py

```diff
@@ -0,0 +1,264 @@
+'''
+
+EXAMPLES = r'''
+- name: Add a new Service Graph
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    service_graph: graph1
+    service_nodes:
+      - type: firewall
+      - type: other
+      - type: load-balancer
+    state: present
+  delegate_to: localhost
+
+- name: Remove a Service Graph
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    service_graph: graph1
+    state: absent
+  delegate_to: localhost
+
+- name: Query a specific Service Graph
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    service_graph: graph1
+    state: query
+  delegate_to: localhost
+
+- name: Query all Service Graphs
+  cisco.mso.mso_schema_template_service_graph:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema1
+    template: Template1
+    state: query
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_service_graph_node_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        service_graph=dict(type='str', aliases=['name']),
+        description=dict(type='str', default=''),
+        display_name=dict(type='str'),
+        service_nodes=dict(type='list', elements='dict', options=mso_service_graph_node_spec()),
+        filter_after_first_node=dict(type='str', choices=['allow_all', 'filters_from_contract']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['service_graph']],
+            ['state', 'present', ['service_graph', 'service_nodes']],
+        ],
+    )
+
+    schema = module.params.get('schema')
+    template = module.params.get('template').replace(' ', '')
+    service_graph = module.params.get('service_graph')
+    display_name = module.params.get('display_name')
+    description = module.params.get('description')
+    service_nodes = module.params.get('service_nodes')
+    filter_after_first_node = module.params.get('filter_after_first_node')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+
+    if filter_after_first_node == 'allow_all':
+        filter_after_first_node = 'allow-all'
+    if filter_after_first_node == 'filters_from_contract':
+        filter_after_first_node = 'filters-from-contract'
```

Comment:
Do you need this outside of present?

Reply:
No. Other modules have it outside present, so following the norm

Reply:
Also change second one to elif